src/lib/hooks/useProfileAvatar.js

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useMemo, useState } from 'react'
+
+import userStorage from '../gundb/UserStorage'
+import { isValidBase64Image } from '../utils/image'
+
+export default (avatar, skipCache = false) => {
+  const cachedBase64 = useMemo(() => {
+    // checking is it base64 data url
+    if (isValidBase64Image(avatar)) {
+      return avatar
+    }
+
+    // if not - return null
+    return null
+  }, [avatar])
+
+  // using cachedState as initial base64 value immediately in the state (if was base64)
+  const [base64, setBase64] = useState(cachedBase64)
+
+  useEffect(() => {
+    // if was base64 set it to state immediately
+    if (cachedBase64) {
+      setBase64(cachedBase64)
+      return
+    }
+
+    // otherwise we're checking is it a valid CID and trying to load it from thes ipfs
+    userStorage
+      .loadAvatar(avatar, skipCache)
```

Comment:
get profile avatar should not be related to user storage, as we display profile avatar of other users in feed.
also you are not using the base64storage to cache other users avatars

Reply:
1. will remove loadAvatar and implement it in this hook
2. base64 storage used for caching all avatars.
- the upload is performing in setAvatar function. 
- Also i added an update script which will upload base64 and store cid in the profile field once each user will log in.
- For the avatars haven't been updated yet i'm falling back to use base64 stored in the GUN as is. 
- As we're storing now CID in the profile field, the CIDs will be also stored & cached in the feed items.
- Base64Storage load method caches avatars by CID. if we have loaded some avatar (despite is it own avatar or belongs to other user) - we won't load it again until we pass skipCache = true param

Reply:
added "on-fly" uploading of the counterpart avatars cached the feed
update script will flush GD_Feed so the fill will be loaded from the GUN and base64 avatars will be converted and uploaded during the scroll